modules/siddhi-core/src/test/java/org/wso2/siddhi/core/window/DelayWindowTestCase.java

```diff
@@ -0,0 +1,417 @@
+import org.wso2.siddhi.core.stream.input.InputHandler;
+import org.wso2.siddhi.core.stream.output.StreamCallback;
+import org.wso2.siddhi.core.util.EventPrinter;
+import org.wso2.siddhi.core.util.SiddhiTestHelper;
+import org.wso2.siddhi.core.util.persistence.InMemoryPersistenceStore;
+import org.wso2.siddhi.core.util.persistence.PersistenceStore;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Delay window implementation testcase
+ */
+
+public class DelayWindowTestCase {
+    private static final Logger log = Logger.getLogger(DelayWindowTestCase.class);
+    private boolean eventArrived;
+    private AtomicInteger count = new AtomicInteger(0);
+    private Long lastValue;
+    private double averageValue;
+    private int removeEventCount;
+    private int inEventCount;
+    private boolean error;
+
+    @BeforeMethod
+    public void init() {
+        count.set(0);
+        eventArrived = false;
+        lastValue = Long.valueOf(0);
+        averageValue = Double.valueOf(0);
+        removeEventCount = 0;
+        inEventCount = 0;
+        error = true;
+    }
+
+    @Test(description = "Check if Siddhi App is created successfully when only one parameter of type either " +
+            "int or long is specified")
+    public void delayWindowTest0() {
+        log.info("DelayWindow Test0 : Testing window parameter definition1");
+        SiddhiManager siddhiManager = new SiddhiManager();
+        String query = "define window eventWindow(symbol string, price int, volume float) delay(2 sec)";
+        SiddhiAppRuntime siddhiAppRuntime = null;
+        try {
+            siddhiAppRuntime = siddhiManager.createSiddhiAppRuntime(query);
+        } catch (SiddhiAppCreationException e) {
+            error = false;
+            log.info(e.getCause());
+        } finally {
+            if (siddhiAppRuntime != null) {
+                siddhiAppRuntime.shutdown();
+            }
+        }
+        AssertJUnit.assertTrue(error);
+    }
+
+    @Test(description = "Check if Siddhi App Creation fails when more than one parameter is included")
+    public void delayWindowTest1() {
+        log.info("DelayWindow Test1 : Testing window parameter definition2");
+        SiddhiManager siddhiManager = new SiddhiManager();
+        String query = "define window eventWindow(symbol string, price int, volume float) delay(2,3) ";
+        SiddhiAppRuntime siddhiAppRuntime = null;
+        try {
+            siddhiAppRuntime = siddhiManager.createSiddhiAppRuntime(query);
+        } catch (SiddhiAppCreationException e1) {
+            error = false;
+            log.info(e1.getCause());
+        } finally {
+            if (siddhiAppRuntime != null) {
+                siddhiAppRuntime.shutdown();
+            }
+        }
+        AssertJUnit.assertFalse(error);
+    }
+
+    @Test(description = "Check if Siddhi App Creation fails when the type of parameter is neither int or long")
+    public void delayWindowTest2() {
```

Comment:
Please use expectedExceptions testng parameter to write these validation test cases rather than asserting error. You can catch and assert the error message too.

Reply:
fixed